### ISSUE TITLE
lr_scheduler unit tests

### ISSUE THREAD

**Issue by tjruwase** [microsoft/DeepSpeed#127]
We need unit tests to cover the following cases
1) scheduler is generating expected lr and mom values corresponding to global steps
2) scheduler is updating optimizer correctly

**Comment by tjruwase** [comment]
Unit tests for expected lr values is handled by PR #339
